CMakeLists.txt

```diff
@@ -14,6 +14,12 @@ project(${SOLUTION_NAME} LANGUAGES CXX VERSION ${SOLUTION_NAME_VERSION})
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_BUILD_TYPE Debug)
+set(CMAKE_CXX_FLAGS "-O3 -Wall -Wextra")
+set(CMAKE_CXX_FLAGS_DEBUG "-O3 -Wall -Wextra")
```

Comment:
Adding `-O3` directly into CMAKE_CXX_FLAGS is not really what you want (you can achieve by running `cmake -DCMAKE_BUILD_TYPE=Release ....`).

For the warning flags you could check the compiler ID (e.g., `if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")`) before adding them.